pkg/actor/resize_pvc.go

```diff
@@ -0,0 +1,280 @@
+}
+
+// Act in this implementation resizes PVC volumes of a CR sts.
+func (rp *resizePVC) Act(ctx context.Context, cluster *resource.Cluster) error {
+	log := rp.log.WithValues("CrdbCluster", cluster.ObjectKey())
+
+	// If we do not have a volume claim we do not have PVCs
+	if cluster.Spec().DataStore.VolumeClaim == nil {
+		log.Info("Skipping PVC resize as VolumeClaim does not exist")
+		return nil
+	}
+
+	// Get the sts and compare the sts size to the size in the CR
+	key := kubetypes.NamespacedName{
+		Namespace: cluster.Namespace(),
+		Name:      cluster.StatefulSetName(),
+	}
+	statefulSet := &appsv1.StatefulSet{}
+	if err := rp.client.Get(ctx, key, statefulSet); err != nil {
+		return errors.Wrap(err, "failed to fetch statefulset")
+	}
+
+	// TODO statefulSetIsUpdating is not quite working as expected.
+	// I had to check status.  We should look at the update code in partition update to address this
+	if statefulSetIsUpdating(statefulSet) {
+		return NotReadyErr{Err: errors.New("resize statefulset is updating, waiting for the update to finish")}
+	}
+
+	status := &statefulSet.Status
+	if status.CurrentReplicas == 0 || status.CurrentReplicas < status.Replicas {
+		log.Info(" resize pvc statefulset does not have all replicas up")
+		return NotReadyErr{Err: errors.New("resize pvc statefulset does not have all replicas up")}
+	}
+
+	// Maybe this should be an error since we should not have this, but I wanted to check anyways
+	if len(statefulSet.Spec.VolumeClaimTemplates) == 0 {
+		log.Info("Skipping PVC resize as PVCs do not exist")
+		return nil
+	}
+
+	clientset, err := kubernetes.NewForConfig(rp.config)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create kubernetes clientset")
+	}
+
+	stsStorageSizeDeployed := statefulSet.Spec.VolumeClaimTemplates[0].Spec.Resources.Requests.Storage()
+	stsStorageSizeSet := cluster.Spec().DataStore.VolumeClaim.PersistentVolumeClaimSpec.Resources.Requests.Storage()
+
+	// If the sizes match do not resize
+	if stsStorageSizeDeployed.Equal(stsStorageSizeSet.DeepCopy()) {
+		log.Info("Skipping PVC resize as sizes match")
+		return nil
+	}
+
+	log.Info("Starting PVC resize")
+
+	// Find all of the PVCs and resize them
+	if err := rp.findAndResizePVC(ctx, statefulSet, cluster, clientset); err != nil {
+		return errors.Wrapf(err, "updating PVCs for statefulset %s.%s", cluster.Namespace(), cluster.StatefulSetName())
+	}
+
+	log.Info("Starting updating sts")
+
+	// Update the STS with the correct volume size, in case more pods are created
+	// We will create a copy and update the copy, and then delete the original without
+	// deleting the Pods.  The new sts is then used to create a new statefulset.
+	if err := rp.updateSts(ctx, statefulSet, cluster); err != nil {
+		return errors.Wrapf(err, "updating statefulset %s.%s", cluster.Namespace(), cluster.StatefulSetName())
+	}
+
+	// TODO this is not working so we will need to patch the sts
```

Comment:
I am wondering a couple of things.  

1. Have RestartClusterOnResize flag, since I think most of the times we do not need to roll the cluster.
2. Yes this code is not working as we need to patch the sts before doing the rolling restart.